testing/e2e/rules.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+function set_git_cached_repo_path() {
+  export TEST_GIT_CACHE_DIR=${TEST_TMPDIR}/git-cache-${WORKSPACE_DIR}
+}
+
+function copy_git_cached_repo_rules_to_workspace() {
+  if [[ -z "${RULES_GIT_DIR:-}" ]]; then
+    export RULES_GIT_DIR=${WORKSPACE_DIR}/rules/git
+  fi
+
+  if [[ ! -d "${RULES_GIT_DIR}" ]]; then
+    mkdir -p -m 0700 "${RULES_GIT_DIR}"
+    cp -r ./rules/git/. ${RULES_GIT_DIR}
+  fi
+
+  cat >${RULES_GIT_DIR}/BUILD <<EOF
+package(default_visibility = ["//visibility:public"])
+
+filegroup(
+    name = "git",
+    srcs = [
+        "BUILD",
+    ],
+)
+EOF
+}
+
+function delete_git_cached_directory() {
+  cache_dir=$1
+  unset RULES_GIT_DIR
+
+  # Clean git cached directory and verify path contains an identifier before deletion
+  if [[ (! -z ${cache_dir}) && -d ${cache_dir} && "${cache_dir}" == *"test-git-cache"* ]]; then
+    rm -rf ${cache_dir}
+  fi
+
+  if [[ -d "${HOME}/.git-cache" ]]; then
+    rm -rf "${HOME}/.git-cache"
```

Comment:
The `git_cache_repository` rule uses a default directory at the home path for storing git indexes if no custom dir supplied. It is a behavior that must get tested so we can't overlook.
Since the testing suite should run within a temporary container, why does it bothers you?